app/views/user_mailer/challenge_assignment_notification.html.erb

```diff
@@ -14,38 +14,38 @@
 
     <% def styled_tag_list(tags) %>
       <% return nil if !tags || tags.empty? %>
-      <% tags.map{|tag| style_link(tag.name, tag_works_url(tag))}.to_sentence.html_safe %>
+      <% tags.map { |tag| style_link(tag.name, tag_works_url(tag)) }.to_sentence.html_safe %>
     <% end %>
 
     <% fandoms = prompt.any_fandom ? "Any" : styled_tag_list(tag_groups["Fandom"]) %>
     <% chars = prompt.any_character ? "Any" : styled_tag_list(tag_groups["Character"]) %>
     <% ships = prompt.any_relationship ? "Any" : styled_tag_list(tag_groups["Relationship"]) %>
-    <% ratings = prompt.any_rating ? "Any" : get_title_string(tag_groups["Rating"]) %>
+    <% ratings = prompt.any_rating ? "Any" : (tag_groups["Rating"] ? get_title_string(tag_groups["Rating"]) : nil) %>
```

Comment:
I was kind of wondering that myself, but not enough to look into it beyond the original bug that presented itself to me on a platter. >.> 

But, in general, this whole thing seems very inconsistent. Any type of tag can have an "Any?" option, and any type of tag can be omitted (unless the mod of a give challenge has decided to make that tag mandatory, of course). I have no idea why they don't all have the same logic I put in here.